core/include/optee_rpc_cmd.h

```diff
@@ -148,6 +148,18 @@
  */
 #define OPTEE_RPC_CMD_BENCH_REG		20
 
+/*
+ * Transfer data in/out via I2C
```

Comment:
well no, this structure is not limiting to 7-bit addresses so I dont think that would be accurate.
And as to having to specify master it is clearly a redundancy...but of course I can do it if it is a condition to get accepted.

IMO you/maintainers have surrounded the few lines of code in this commit with an unnecessary amount of redundant comments...I think it is a mistake but please let me know what needs doing so this gets merged.

Reply:
Ok the ABI does not limit to 7bit, but the REE side needs to know whether to use 7bit or 10bit address mode. If you expect/implement the 7bit mode on REE side, it lloks simpler to state this command is designed for 7bit. An extra 10bit address mode command can be later added if needed.

As for `master mode`, maybe that's not a wording found in I2C literature, feel free to reword.

Reply:
Clear and concise is the motto. It should be possible to understand this without consulting you afterwards. If you can word the description better please do, in fact it is preferable if the one writing the code comments it himself.

Reply:
but precisely for that it would be incorrect to declare it here: the actual dependency with the 7bit length is not defined here (in the command) but in rpc_io_i2c_transfer as already documented btw . 

So doing it here is not only unnecessary but also not correct: we will have to edit this file - when we add support to 10 bits in rpc_io_i2c_transfer.c - just to change the comment but not the code (and isn't the code supposed to do what the comment clarifies?...I would think twice about a file that requires comments to be modified but not their source code because something changed on some other file in the tree....a strong smell that something is not properly abstracted.

anyway...sorry for the explanation ...let me just make the changes and push...it will save us time.

Reply:
> ...not properly abstracted.

I don't understand, are you not satisfied with the abstraction in your patch? Or is there something else that should be improved?

Reply:
I am satisfied with the patch - just not with the comment that I had to add to optee_rpc_cmd.h  to announce a dependency which is not applicable to this file (but to rpc_io_i2c instead).

meaning that once we add 10 bit support in rpc_io_i2c.c we will have to remove a comment from optee_rpc_cmd.h

Reply:
@ldts, I don't catch you point. The REE I2C needs to know if 7bit or 10bit address is to be used.
If this RPC func ID is designed for 7bit, let's state it.
This header file should be the reference for the I2C RPC ABI.
You/we will no be able to reuse this RPC func ID for both 7bit and 10bit address cases without changing its ABI.

Reply:
I dont understand, why wont we be able to reuse this ID for 10bit addresses? cant each value hold up to 64bits?

Reply:
Field `chip` is large enough for a 10bit address but REE RPC handler as no way to know if it shall treat this field as a 7bit or a 10bit address value.

Reply:
that will  not matter - at least not in Linux AFAIK (7 bit addresses are stored in the lower 7 bits of a 16 bit field); I cant speak for other REEs but IMO it would be weird having to add a flag to distinguish between 7 and 10 bit address....would make the core stack ugly

This is the Linux side of the RPC call for reference

	client.addr = params[0].u.value.c;

	client.adapter = i2c_get_adapter(params[0].u.value.b);
	if (!client.adapter)
		goto bad;

	snprintf(client.name, I2C_NAME_SIZE, "i2c%d", client.adapter->nr);

	buf = params[1].u.memref.shm->kaddr;
	len = params[1].u.memref.size;

	switch (params[0].u.value.a) {
        case OPTEE_MSG_RPC_CMD_I2C_TRANSFER_RD:
		ret = i2c_master_recv(&client, buf, len);
		break;
	case OPTEE_MSG_RPC_CMD_I2C_TRANSFER_WR:
		ret = i2c_master_send(&client, buf, len);
		break;
	default:
		i2c_put_adapter(client.adapter);
		goto bad;
	}

Reply:
> Clear and concise is the motto. It should be possible to understand this without consulting you afterwards. If you can word the description better please do, in fact it is preferable if the one writing the code comments it himself.

I would have no trouble adding myself as maintainer of this section of code if that helps.

Reply:
So for 10 bit mode the lower 10 bits (of `params[0].u.value.c`) carry the 10bit address and the rest are zero (`MBZ` or must be zero).
In 7 bit mode the lower 7 bits carry the 7bit address and the rest are zero.

Is that enough or is a flag needed too in order to explicitly tell if it's a 10bit or 7bit address?

If you could update the ABI description to cover this I think this should be good enough to handle eventual future extensions. I guess this is what the discussion here has been about.

Reply:
I'll add support for 10 bit addresses on linux. will be better.

Reply:
(didnt answer your question: we could reuse and use lower 16 for address and higher 16 for flags - those are the lengths in linux). but since there is no real need I'll just add another THREAD_PARAM_VALUE indicating a 10 bit address. if in the future we need to support  more flags, we can add to this thread param and will look cleaner IMO